### ISSUE TITLE
Zombies 'teleporting'

### ISSUE THREAD

**Issue by AzTheProtoGoat** [CleverRaven/Cataclysm-DDA#60861]
### Describe the bug

So nowadays while fighting zombies, i noticed that some of them sort of 'teleport' one tile or couple forward/in other directions. So they get an extra movement in the same time as usual

### Steps to reproduce

Kind of hard to reproduce as it takes a lot of time, i started a recording and fought zombies in a city in groups, then i seperated one of them and made him walk around to me, when he got close enough he usually does the leap, had a try on my debug save with it and a fat zombie did the same thing too, but they didnt become instantly adjacent to me.

### Expected behavior

I expected the zombie to move normally instead of teleporting 1 tile forward

### Screenshots

Im gonna put a video here as its hard to explain it via screenshots

https://youtu.be/P9DoQkFvNqo

To the fat zombie mentioned : Different save and a bit different mods

https://youtu.be/GanIg97gvHY

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1889 (21H2)
- Game Version: b1fb6f2 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Xedra Evolved [xedra_evolved],
    DinoMod [DinoMod],
    Extra Mutated Scenarios [extra_mut_scens],
    Mining Mod [Mining_Mod],
    sees-player icon, HitButton_iso [sees_player_hitbutton],
    sees-player, run, crouch, bleed icons, retrodays [sees_player_retro],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Tamable Wildlife [Tamable_Wildlife],
    Blaze Industries [blazeindustries],
    <color_light_blue>Lighter Weariness</color> [light_weariness],
    No Hope [no_hope],
    Make bows great again [bow_opness],
    Magiclysm [magiclysm]
]

**Comment by Bulletfang** [comment]
I've noticed this too, started happening quite a while ago but I chalked it up to the zombies movement being erratic and just randomly being faster for one turn as they stumble forward.

**Comment by AzTheProtoGoat** [comment]
Still i doubt that's an intended behaviour for them, considering there is no real indicator of it happening anywhere in the log during combat

**Comment by snipercup** [comment]
I've seen this too and I think fat zombies shouldn't leap tiles considering their low speed.

**Comment by AzTheProtoGoat** [comment]
It's also something that did not happen before in my experience. It only started to happen recently

**Comment by LyleSY** [comment]
I wonder if this is the STUMBLE code being strange. It's supposed to make zombie movement erratic but I don't think it's supposed to give them (more) super powers

**Comment by propurty** [comment]
I remember a PR talking about stumbling zombies being added. Which I assume would look like them teleporting if they suddenly skipped forward a tile or got very fast for a split second. I think it also mentioned the zombies being downed when it happens. Not really sure but maybe someone can find the PR mentioning it.

**Comment by Zireael07** [comment]
Stumbling seems to be an older feature than I thought (I tried to find the recent PR @propurty and I recall, but nope, but I found a PR on the topic as early as 2016: #15705 )

**Comment by propurty** [comment]
@Zireael07 I can't find it either. A quick Google search led me to the wiki. While I'm sure it's outdated, it does mention stumbling as a feature of creatures and zombies. It says that they stop stumbling 4 tiles away from a target. Target can be an enemy or sound. 

Again assume it's outdated. Wish I could find more info on stumbling itself, but I just don't have the time at the moment.

**Comment by AzTheProtoGoat** [comment]
> It says that they stop stumbling 4 tiles away from a target. Target can be an enemy or sound.
Yeah that's def outdated, cause in both of the videos the stumbling happened when there was only like 3 tile gap between me and a monster

**Comment by Sutremaine** [comment]
monmove.cpp

            // A flag to allow non-stumbling critters to stumble when the most direct choice is bad.
            bool bad_choice = false;

And then a bunch of conditions.
Somewhere below that there's also this thing:

    // Finished logic section.  By this point, we should have chosen a square to move to (moved = true).
    if( moved ) { // Actual effects of moving to the square we've chosen
[snip snip]
 } else {
        moves = 0;
        stumble();
        path.clear();
    }

Not sure what stumble() is doing exactly, but it might be another way for stumbles to happen without the creature being a stumbler or using the bad_choice flag.

**Comment by kevingranade** [comment]
Closing this because there's no way to reproduce it and now the videos are gone and the speculation about it is definitely wrong.
1. monster::stumble() spends moves just like regular movement.
2. Everywhere that calls monster::stumble() ALSO sets monster moves to 0

If someone has a workable angle on tracking this down, please feel free to open a new issue, but this one doesn't leave us with anything to go on.

It's also possible that this was happening at some point but some change made it stop.
